File: wind-log/font/src/services/star.js
import request from '@/utils/request'

export const fetchIsStar = (userId, articleId) => {
    return request(`http://localhost:8080/is_star/${userId}/${articleId}`)
}

export const starArticle = (star) => {
    return request(`http://localhost:8080/star_article`, {
        method: 'post',
        data: star
    })
}

export const cancleStar = (star) => {
    return request(`http://localhost:8080/cancel_star`, {
        method: 'delete',
        data: star
    })
}

export const fetchStarCategory = (userId) => {
    return request(`http://localhost:8080/star_category/${userId}`)
}

File: wind-log/back/src/main/java/com/zs/windlog/Do/Star.java
package com.zs.windlog.Do;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Star {
    private Integer starId;
    private Integer articleId;
    private Integer userId;
    private String starCategory;
}
